src/gossip/service.zig

```diff
@@ -1882,36 +1880,17 @@ pub const GossipService = struct {
                 .NodeInstance => {},
                 else => {
                     // only allow values where both the sender and origin match our shred version
-                    if (!sender_matches or
-                        !gossip_table.checkMatchingShredVersion(gossip_value.id(), my_shred_version))
-                    {
-                        const removed_value = gossip_values_array.swapRemove(i);
-                        bincode.free(self.gossip_value_allocator, removed_value);
-                        continue; // do not incrememnt `i`. it has a new value we need to inspect.
+                    if (!gossip_table.checkMatchingShredVersion(gossip_value.id(), my_shred_version)) {
+                        continue; // do not increment `i`, this wasn't valid.
```

Comment:
dont think this logic is correct anymore - we still need to swapRemove to make sure the values we insert have a matching shred version

Reply:
You may be right, I will look into this.

Reply:
So yeah, I mistook the purpose of the code originally. I've now implemented a much more efficient Hoare's partitioning scheme to sort these. It's `O(N)` instead of `O(N^2)` like the other one was.